yang/frr-logging.yang

```diff
@@ -0,0 +1,197 @@
+
+  revision 2025-06-07 {
+    description "Initial revision.";
+    reference "FRRouting";
+  }
+
+  typedef syslog-format {
+    type enumeration {
+      enum rfc5424 {
+        description "RFC5424 syslog format.";
+      }
+      enum rfc3164 {
+        description "RFC3164 syslog format.";
+      }
+      enum local-syslogd {
+        description "Local syslogd format without hostname.";
+      }
+      enum journald {
+        description "systemd journald native format.";
+      }
+    }
+    description "Log message formatting.";
+  }
+
+  container logging {
+    description "Logging configuration.";
+
+    container stdout {
+      presence "Enable stdout logging";
+      description "Logging to stdout.";
+      leaf level {
+        type syslogtypes:severity;
+        default debug;
+        description "Lowest level for which to log to stdout.";
+      }
+    }
+
+    container syslog {
+      presence "Enable syslog logging";
+      description "Logging to syslog.";
+      leaf level {
+        type syslogtypes:severity;
+        default debug;
+        description "Lowest level for which to log to syslog.";
+      }
+    }
+
+    container file {
+      description "Logging to a file.";
+      leaf filename {
+        type string;
+        description "Log filename.";
+      }
+      leaf level {
+        type syslogtypes:severity;
+        default debug;
+        description "Lowest level for which to log to the file.";
+      }
+    }
+
+    container filtered-file {
+      description
+        "Log messages to a file, filtering out messages that match any of the
+         filter-text entries.";
+      leaf filename {
+        type string;
+        description "Log filename.";
+      }
+      leaf level {
+        type syslogtypes:severity;
+        default debug;
+        description "Lowest level for which to log to the filtered file.";
+      }
+    }
+
+    leaf-list filter-text {
+      type string;
+      description "Log message filters.";
+    }
+
+    list daemon-file {
+      key daemon;
+      description "Per-daemon log files.";
+      leaf daemon {
+        type string;
+        description "Name of the daemon.";
+      }
+      leaf filename {
+        type string;
+        mandatory true;
+        description "Log filename.";
+      }
+      leaf level {
+        type syslogtypes:severity;
+        default debug;
+        description "Lowest level for which to log to the daemon log file.";
+      }
+    }
+
+    leaf facility {
+      type identityref {
+        base syslogtypes:syslog-facility;
+      }
+      default syslogtypes:daemon;
+      description "Syslog facility.";
+    }
+
+    leaf record-priority {
+      type boolean;
+      default false;
+      description "Include priority in non-syslog messages.";
+    }
+
+    leaf record-severity {
+      type boolean;
+      default false;
+      description "Include priority in non-syslog messages.";
```

Comment:
We maybe use priority and severity interchangeably, but did you mean `severity` here?

Reply:
yes, should be `Include severity ...`